reference/basic-architecture.md

```diff
@@ -9,61 +9,66 @@ More detailed guides to Mayastor's components, their design and internal structu
+### csi-node
 
-The scheduling of Mayastor pods is determined declaratively by using a DaemonSet specification. By default, a `nodeSelector` field is used within the pod spec to select all worker nodes to which the user has attached the label `openebs.io/engine=mayastor` as recipients of a Mayastor pod. It is in this way that the MayastorNode count and location is set appropriate to the hardware configuration of the worker nodes \(i.e. which nodes host the block storage devices to be used\), and the capacity and performance demands of the cluster.
+    The csi-node pods within a cluster implement the node plugin component of Mayastor's CSI driver. As such, their function is to orchestrate the mounting of Mayastor-provisioned volumes on worker nodes on which application pods consuming those volumes are scheduled. By default, a csi-node pod is scheduled on every node in the target cluster, as determined by a DaemonSet resource of the same name. Each of these pods encapsulates two containers, csi-node, and csi-driver-registrar.
 
-### Mayastor-CSI
+    The node plugin does not need to run on every worker node within a cluster and this behavior can be modified, if desired, through the application of appropriate node labeling and the addition of a corresponding nodeSelector entry within the pod spec of the csi-node DaemonSet. It should be noted that if a node does not host a plugin pod, then it will not be possible to schedule a pod on it which is configured to mount Mayastor volumes.
 
-The mayastor-csi pods within a cluster implement the node plugin component of Mayastor's CSI driver. As such, their function is to orchestrate the mounting of Maystor provisioned volumes on worker nodes on which application pods consuming those volumes are scheduled. By default a mayastor-csi pod is scheduled on every node in the target cluster, as determined by a DaemonSet resource of the same name. These pods each encapsulate two containers, `mayastor-csi` and `csi-driver-registrar`
+### etcd
+   
+    etcd is a distributed reliable key-value store for the most critical data of a distributed system. Mayastor uses etcd as a reliable persistent store for its configuration and state data.
 
-It is not necessary for the node plugin to run on every worker node within a cluster and this behaviour can be modified if so desired through the application of appropriate node labeling and the addition of a corresponding `nodeSelector` entry within the pod spec of the mayastor-csi DaemonSet. It should be noted that if a node does not host a plugin pod, then it will not be possible to schedule pod on it which is configured to mount Mayastor volumes.
+### Supportability CLI:
```

Comment:
What's the "Supportability CLI"?

Support bundles and other diagnostic information are obtained through the same kubectl plugin as is used to administer the other functions of Mayastor.  There's no separate CLI for supportability.